RoadmapChecklist/RoadmapChecklist.Api/Controllers/UserController.cs

```diff
@@ -19,14 +19,17 @@ public UserController(IUserService userService)
         [HttpPost("register")]
         public IActionResult Register([FromBody]Register userRegisterModel)
         {
-            var isUserExist = service.IsUserExist(userRegisterModel.Email.ToLower(), userRegisterModel.UserName).Data;
-            if (isUserExist != null)
+            var isUserValidForRegister = service.IsUserValidForRegister(userRegisterModel.Email.ToLower(), userRegisterModel.UserName);
+
+            var user = isUserValidForRegister.IsSuccess ? isUserValidForRegister.Data : null;
+            
+            if (isUserValidForRegister != null)
```

Comment:
Null kontrolünü yapmak için biraz geç değil mi sence de? :) @handeebrar

Reply:
Bu kod Outdated kalmış @aykutonen. Öncesinde de dediğim gibi burada dikkat çekmeye çalıştığın problemi anlamadım. Kullanıcının register olabilmesi için daha önceden username ve mail adresi kullanılmıyor olmalı. Service katmanından bu bilgiyi alıp result.data'nın null kontrolünü yapıyoruz. Farklı bir şey söylemek istediysen tekrar bakabilirim.